autoPyTorch/pipeline/components/base_component.py

```diff
@@ -203,14 +219,22 @@ def check_requirements(self, X: Dict[str, Any], y: Any = None) -> None:
         if y is not None:
             warnings.warn("Provided y argument, yet only X is required")
 
-        for requirement in self._fit_requirements:
-            if requirement.name not in X.keys():
-                raise ValueError("To fit {}, expected fit dictionary to have '{}'"
-                                 " but got \n {}".format(self.__class__.__name__, requirement.name, list(X.keys())))
-            elif not isinstance(X[requirement.name], requirement.type):
-                raise TypeError("Expected {} to be instance of {} got {}".format(requirement.name,
-                                                                                 requirement.type,
-                                                                                 type(X[requirement.name])))
+        if self._fit_requirements:
```

Comment:
Better `if self._fit_requirements is not None`, or even better initialize `self._fit_requirements: Optional[List[FitRequirement]] = list()` in line 95 and delete the if statement